.github/workflows/build-test-lint.yml

```diff
@@ -0,0 +1,37 @@
+name: Ruby Build, Lint and Test
+
+on:
+    push:
+
```

Comment:
Hey @vigneshk-tw 

Could you please change the trigger condition to PR.

  pull_request:
    branches:
      - master